python/GafferSceneUI/MeshTangentsUI.py

```diff
@@ -36,51 +36,102 @@
 
 import Gaffer
 import GafferScene
+import IECoreScene
 
 Gaffer.Metadata.registerNode(
 
 	GafferScene.MeshTangents,
 
 	"description",
 	"""
-	Adds surface tangent primitive variables to the mesh using the specified positions and UVSet. 
+	Adds surface tangent primitive variables to the mesh using the specified positions and UVSet.
 	""",
 
+	"layout:activator:uvActivator", lambda parent : parent["mode"].getValue() == int(IECoreScene.MeshAlgo.MeshTangentsMode.UV),
+	"layout:activator:leftHandedActivator", lambda parent : parent["orthogonal"].getValue() == True,
+
 	plugs = {
 
-		"uvSet" : [
+		"mode" : [
+
 			"description",
 			"""
-			Name of the UV set primitive variable used to calculate uTangent & vTangent.
+			The style of how to calculate the Tangents.
+			(UV) calculates the tangents based on the gradient of the the corresponding UVs
+			(FirstEdge) defines the vector to the first neighbor as tangent and the bitangent orthogonal to tangent and normal
+			(TwoEdges) defines the vector between the first two neighbors as tangent and the bitangent orthogonal to tangent and normal
+			(PrimitiveCentroid) points the tangent towards the primitive centroid and the bitangent orthogonal to tangent and normal
 			""",
+
+			"preset:UV", IECoreScene.MeshAlgo.MeshTangentsMode.UV,
+			"preset:FirstEdge", IECoreScene.MeshAlgo.MeshTangentsMode.FirstEdge,
+			"preset:TwoEdges", IECoreScene.MeshAlgo.MeshTangentsMode.TwoEdges,
+			"preset:PrimitiveCentroid",IECoreScene.MeshAlgo.MeshTangentsMode.PrimitiveCentroid,
+
+			"plugValueWidget:type", "GafferUI.PresetsPlugValueWidget",
+
+			"layout:index", 4,
+		],
+
+		"orthogonal" : [
+			"description",
+			"""
+			Adjusts vTangent to be orthogonal to the uTangent.
+			""",
+			"layout:index", 5,
+		],
+
+		"leftHanded" : [
+			"description",
+			"""
+			Make the local coordinate frame left handed
+			""",
+			"layout:activator", "leftHandedActivator",
+			"layout:index", 6,
 		],
 
 		"position" : [
 			"description",
 			"""
-			Name of the primitive variable which contains the position data used calculate tangents & binormals. 
+			Name of the primitive variable which contains the position data used calculate tangents & binormals.
 			For example 'Pref' would compute tangents using the reference positions (if defined)
 			""",
+			"layout:section", "Settings.Input",
 		],
 
-		"orthogonal" : [
+		"normal" : [
+			"description",
+			"""
+			Name of the primitive variable which contains the normal vector jdata used calculate tangents & binormals.
+			""",
+			"layout:section", "Settings.Input",
```

Comment:
If I understand correctly, we should add something to grey out the normal field when in UV mode?